src/gui/torrentcontentmodel.cpp

```diff
@@ -88,6 +96,40 @@ namespace
             return QIcon(iconPixmap);
         }
     };
+#elif defined(Q_OS_MAC)
+    // There is a similar bug on macOS, to be reported to Qt
+    // https://github.com/qbittorrent/qBittorrent/pull/6156#issuecomment-316302615
+    class MacFileIconProvider: public UnifiedFileIconProvider
+    {
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            QSize sz(32, 32);
+            QPixmap pixmap = pixmapForExtension(info.suffix(), sz);
```

Comment:
> Well, in my opinion it should be up to the OS to decide what to pass for a an empty extension…

Agree, but, as you mentioned, we already use a custom icon for that. So I wanted us to be consistent.

> …and I wanted to use completeSuffix initially, but it reads the extension as the first dot in a file. In most cases it will be useless,…

Why will it be useless? MacOS does not assign special file types to such files?